### PR TITLE
Create CITATION.cff

### PR DESCRIPTION
**Pull request by tlubowe**
just added this in cuQuantum - might be useful for you

### CODE REVIEW

**Comment by tlubowe** [comment]
I have read the Contributor License Agreement and I hereby accept the Terms.

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,12 @@
+cff-version: 1.2.0
+title: 'CUDA Quantum'
+message: 'If you use this software, please cite it as below.'
+authors:
+  - name: "The CUDA Quantum development team"
+license: Apache-2.0
+license-url: "https://github.com/NVIDIA/cuda-quantum/blob/main/LICENSE"
+repository-code: "https://github.com/nvidia/cuquantum"
```
Comment (amccaskey) [inline CITATION.cff:8]: @tlubowe Would you update this to cuda-quantum?

Comment (tlubowe) [reply CITATION.cff]: updated - good catch

**Review by bettinaheim — APPROVED** [review]

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,12 @@
+cff-version: 1.2.0
```
[the remark this answers is not in the document]
Comment (tlubowe) [reply CITATION.cff:1]: No, thats the version of cff used. There's a commented out `version` at the bottom you can use to denote sw version for the library citation. This is where where we say cuQuantum is version 23.03 for example.

[the remark this answers is not in the document]
Comment (tlubowe) [reply CITATION.cff:1]: I can modify it to include 0.3.0 if you would like to list the CUDA Quantum version

[the remark this answers is not in the document]
Comment (tlubowe) [reply CITATION.cff:1]: we can add the zenodo doi number at the bottom of this .cff as well. I'm happy to do that if you like just let me know. Then whenever you're ready for release it should pick it up like you said.

**Pull request merged**